submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,126 @@
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+In this proposal, we explore two options to make Submariner work with OVN IC. More details about the upgrade and other design details shall
+be added once agree on one of them.
+
+### Option1: Add extra routes in existing Architecture
+
+ This proposal makes minimal changes in the network plugin syncer to get OVN IC working. With this, we add new routes in the OVN cluster
+router.
+
+![UpdatedTopology](./images/submariner-ovn-ic.png)
+
+These are the changes that will be needed
+
+1) Network plugin Syncer needs to be `zone` aware
+2) In the `zone` where Submariner Gateway is present an extra route will be added to the `ovn-cluster-router` to direct the traffic to remote CIDRs connected via
+submariner via the Submariner router. This is to facilitate inter-cluster connectivity for traffic generated from other zones of the cluster.
+3) In all the other zones a route will be added to direct the traffic to remote cluster CIDRs, connected via submariner, to the transit
+switch that connects to the gateway node
+
+#### Node Listener
+
+A new node listener shall be added in NetworkPluginSyncer.
+
+* The node listener shall be responsible for reading the node labels/annotations and identifying the zone in which the Submariner gateway is
+deployed and the transit switch IP for it.
+* When a new node is added or a node joins a zone, the node listener needs to call the NetworkPluginSyncer handler to program the required
+rules. It will also remove any existing stale rules.
+* When a node leaves no action is needed
+* When the gateway nodes join a new zone, leaving the current one, all the switches will need to be reprogrammed as the transit switch IP
+can change.
+
+#### Network Plugin Syncer
+
+Based on the info retrieved by the Node listener, the network plugin syncer can provision the Submariner infrastructure and program the
+rules in the OVN database.
+
+* If no zone is detected(will be required for backward compatibility) or a global zone is detected the Submariner infrastructure and flows
+will remain as it is today.
+
+* If a zone configuration is detected
+
+1) In the gateway node zone a new routing rule can be added to forward the traffic destined for Submariner remote CIDRs
+to the Submariner router. This is required for the traffic coming from pods in a node that are in a different zone.
+
+2) In the non-gateway zones a route will be added for the traffic destined for remote cluster CIDR to forward it to the gateway node
```

Comment:
What should happen in a GW switchover ?

Reply:
This current g/w switch over logic should be able to handle it, depends on the handlers we add the new routes.

Reply:
Maybe it is worth elaborating on the required changes (if any) to support GW switchover